tests/agents/gherkin-specs/user_agent.feature

```diff
@@ -0,0 +1,19 @@
+Feature: Agent Transport User agent Header
+
+  Scenario: Default user-agent
+    Given an agent
+    When service name is not set
+    When service version is not set
+    Then the User-Agent header matches regex 'elasticapm-[a-z]+/[^ ]*'
```

Comment:
Should these regexes start with '^' to match the prose `Header value should **start** with ...`?

Reply:
Good point, I'll add it now.